protocol/x/revshare/keeper/revshare.go

```diff
@@ -76,3 +76,30 @@ func (k Keeper) CreateNewMarketRevShare(ctx sdk.Context, marketId uint32) {
 	}
 	k.SetMarketMapperRevShareDetails(ctx, marketId, details)
 }
+
+func (k Keeper) GetMarketMapperRevenueShareForMarket(ctx sdk.Context, marketId uint32) (
+	address sdk.AccAddress,
+	revenueSharePpm uint32,
+	err error,
+) {
+	// get the revenue share details for the market
+	revShareDetails, err := k.GetMarketMapperRevShareDetails(ctx, marketId)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// check if the rev share details are expired
+	if revShareDetails.ExpirationTs < uint64(ctx.BlockTime().Unix()) {
+		return nil, 0, nil
+	}
+
+	// Get revenue share params
+	revShareParams := k.GetMarketMapperRevenueShareParams(ctx)
```

Comment:
non-blocking: Just for safety, wonder if we should just cap the `ppm` parameter at 1,000,000.